examples/acnn.py

```diff
@@ -0,0 +1,247 @@
+import deepchem as dc
+import json
+import numpy as np
+
+from copy import deepcopy
+from hyperopt import hp, fmin, tpe
+from shutil import copyfile
+
+from utils import init_trial_path, load_dataset, EarlyStopper
+
+
+def load_model(save_path, args, tasks, hyperparams):
+  if args['dataset'] in ['PDBbind']:
+    mode = 'regression'
+    n_classes = None
+  else:
+    raise ValueError('Unexpected dataset: {}'.format(args['dataset']))
+
+  if args['featurizer'] == 'AC':
+    f1_num_atoms = 100
+    f2_num_atoms = 1000
+    c_num_atoms = f1_num_atoms + f2_num_atoms
+    max_num_neighbors = 12
+
+  if args['model'] == 'ACNN':
+    model = dc.models.AtomicConvModel(
+        n_tasks=len(tasks),
+        frag1_num_atoms=f1_num_atoms,
+        frag2_num_atoms=f2_num_atoms,
+        complex_num_atoms=c_num_atoms,
+        max_num_neighbors=max_num_neighbors,
+        layer_sizes=hyperparams['layer_sizes'],
+        dropout=hyperparams['dropout'],
+        learning_rate=hyperparams['lr'],
+        batch_size=12,
+        model_dir=save_path,
+    )
+  else:
+    raise ValueError('Unexpected model: {}'.format(args['model']))
+
+  return model
+
+
+def main(save_path, args, hyperparams):
+  # Dataset
+  args, tasks, all_dataset, transformers = load_dataset(args)
+  train_set, val_set, test_set = all_dataset
+
+  # Metric
+  if args['metric'] == 'roc_auc':
+    metric = dc.metrics.Metric(dc.metrics.roc_auc_score, np.mean)
+  elif args['metric'] == 'rmse':
+    metric = dc.metrics.Metric(dc.metrics.rms_score, np.mean)
+  else:
+    raise ValueError('Unexpected metric: {}'.format(args['metric']))
+
+  all_run_val_metrics = []
+  all_run_test_metrics = []
+
+  for _ in range(args['num_runs']):
+    # Model
+    model = load_model(save_path, args, tasks, hyperparams)
+    # Object for early stop tracking
+    stopper = EarlyStopper(save_path, args['metric'], args['patience'])
+
+    # 100 for maximum number of epochs
+    for epoch in range(100):
```

Comment:
Should we make the number of epochs a parameter?

Reply:
I kept this to be consistent with the GNN script, and because with early stopping the training never actually hits 100 epochs.